packages/html-reporter/src/headerView.tsx

```diff
@@ -47,7 +50,7 @@ export const HeaderView: React.FC<React.PropsWithChildren<{
       <form className='subnav-search' onSubmit={
         event => {
           event.preventDefault();
-          navigate(`#?q=${filterText ? encodeURIComponent(filterText) : ''}`);
+          navigate(`#?q=${filterText ? decodeURIComponent(filterText) : ''}`);
```

Comment:
nit: i'm not sure we need to encode/decode it here at all since it is a part of the anchor

Reply:
encodeURIComponent is not correct method to use it
```
   Expected pattern: /@smoke/
    Received string:  "http://localhost:50414/#?q=%40smoke"
```

Reply:
We parse hash as follows: ```const searchParams = new URLSearchParams(window.location.hash.slice(1));```, so if the value of `filterText` is `foo&bar=baz`, you end up with `q==foo`, which is not expected.